### PR TITLE
Added Project Documentation and EDA notebooks

### PR DESCRIPTION
**Pull request by aakankshaduggal**
## Related Issues and Dependencies

Related to issue #1, provides the documentation and initial analysis for the project - Cloud Price Analysis

## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

Adds the Project Documentation for Project - Cloud Price Analysis and initial analysis notebooks for the same.

### CODE REVIEW

File: docs/Forecasting.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+*   **Manage Expenses :** Knowing the forecast for the end of the month can help operators to understand if everything remains as expected and take quick action if resources and costs get out of hand.
+*   **Improve Budgeting Process :** End of quarter and end of year forecasts can help improve the budgeting  process and provide support for the financial department.
+*   **Improve Capability of Cost Management :** Being able to use the forecast as a basis can improve the capability of cost management to show true costs when BOM rating needs to be done.
+
+
+### A. Problem Statement:
+
+_In as direct terms as possible, provide the “Data Science” problem statement version of the overview. Think of this as translating the above into a more technical definition to execute on._
+
+The aim of this project is to come up with a linear model that assumes a relationship between the input variable (month) and the predicted variable (Cost). Cost needs to be calculated from a linear combination of the input variable (month).
```
Comment (MichaelClifford) [inline docs/Forecasting.md:39]: Is developing a linear model the goal here? Or is it to generate an accurate forecast of costs based on current and previous cloud compute usage? Above you note that the initial approach will be a linear regression, but I'm not sure if you want to define that as the aim. You could updated it here to be a bit more specific, outlining the things you've already done in your forecasting notebook, like, "start by using a linear regression model on the existing timeseries data set that provides costs to project what those costs might be in the next 3 months" or something like that.

File: docs/Forecasting.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+### B. Checklist for project completion
+
+_Provide a bulleted list of the concrete deliverables and artifacts that, when complete, define the completion of the project._
+
+1. Linear Regression on Dummy Data
```
Comment (MichaelClifford) [inline docs/Forecasting.md:45]: "Linear Regression on Sales Data" , or explain why Dummy data is needed here

File: docs/Forecasting.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+*   Red Hat Cloud Spending data for last two months.
+
+### References
+
+1. [Cost management forecasting and optimization with AI](https://docs.google.com/document/d/1hP3gMkQ6jSPNzDAoCC_QG6WNekJECXDZllxpko7Q924/edit?usp=sharing)  by Sergio Ocón-Cárdenas.
+
+### Github
+
+[https://github.com/aicoe-aiops/cloud-price-analysis](https://github.com/aicoe-aiops/cloud-price-analysis)
```
Comment (MichaelClifford) [inline docs/Forecasting.md:59]: Are these public?

**Review by MichaelClifford — CHANGES_REQUESTED** [review]

**Comment by MichaelClifford** [comment]
@aakankshaduggal were you able to get the Sesheta pre-commit test to pass locally?

File: docs/Cloud-price-analysis.adoc [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    ... Do the same visually for the full price list coming from different vendors (it is not necessary to be able to compare between vendors in this phase)
+. *Use based comparison:* Big customers usually have their own price lists, and thus the generic optimization can be misleading. Repeating the results with a customer specific price list would allow the customer to get additional value from the use cases.
+  .. The price list will be a delta on top of the basic use case, where adjacencies are the same, but prices are different and specific to the customer.
+. *Comparison through time:* AWS, Azure and other clouds change the price list every few days. Identifying new additions and changes to the price list (i.e drift analysis) would make it easier for a customer to understand what is happening and take actions as a delta of the existing status quota.
```
[the remark this answers is not in the document]
Comment (aakankshaduggal) [reply docs/Cloud-price-analysis.adoc:48]: This was specified by the Cost Management Team. I have removed "Drift Analysis" for now.

File: docs/Cloud-price-analysis.adoc [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+* Find unused and unattached resources
+* Identify and consolidate idle resources
+* Analyze the current usage
+* Use graphs(like heatmaps) to identify the usage density
+* Optimize the Reserved Instances/Spot Instances usage
+
+. *Impact*
+
+The completion of this project means AI CoE has a tool at the disposal of other Red Hat teams to use to predict the optimal way of managing cost.
```
[the remark this answers is not in the document]
Comment (aakankshaduggal) [reply docs/Cloud-price-analysis.adoc:66]: This looks good :+1:

File: docs/Cloud-price-analysis.adoc [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+==== A. Problem Statement:
+
+_In as direct terms as possible, provide the “Data Science” problem statement version of the overview. Think of this as translating the above into a more technical definition._
+
+Given  a dataset that talks about the cloud needs of RedHat, build a Cost-Optimization model  that allows the user to make a wise decision on how cloud services should be managed with time. The model should be configurable, automated and should display detailed information behind each prediction.
```
[the remark this answers is not in the document]
Comment (aakankshaduggal) [reply docs/Cloud-price-analysis.adoc:72]: Sure, on it

File: docs/Cloud-price-analysis.adoc [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+
+==== Data Sets
+
+* Aws cli and API (https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/using-ppslong.html)
+* Price list (public)
+* Pricing web page (https://aws.amazon.com/pricing/)
+
+==== References
+
+. https://docs.google.com/document/d/1hP3gMkQ6jSPNzDAoCC_QG6WNekJECXDZllxpko7Q924/edit?usp=sharing[Cost management forecasting and optimization with AI]  by Sergio Ocón-Cárdenas.
```
[the remark this answers is not in the document]
Comment (aakankshaduggal) [reply docs/Cloud-price-analysis.adoc:131]: Nope, I'll remove this

File: docs/Cloud-price-analysis.adoc [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+. https://link.springer.com/content/pdf/10.1007/s13174-011-0032-0.pdf[https://link.springer.com/content/pdf/10.1007/s13174-011-0032-0.pdf] Paper for cost Optimization
+
+==== Gitlab
+
+. https://gitlab.cee.redhat.com/aduggal/modeling/-/tree/master/use_cases/cloud-price-analysis[https://gitlab.cee.redhat.com/aduggal/modeling/-/tree/master/use_cases/cloud-price-analysis]
```
[the remark this answers is not in the document]
Comment (aakankshaduggal) [reply docs/Cloud-price-analysis.adoc:136]: I'll remove this as well

**Comment by aakankshaduggal** [comment]
> @aakankshaduggal were you able to get the Sesheta pre-commit test to pass locally?

Yes

**Comment by aakankshaduggal** [comment]
@MichaelClifford I have implemented the suggested changes.

**Review by tumido — APPROVED** [review]
:+1: I like it! @MichaelClifford any objections or pending change requests?

**Review by MichaelClifford — COMMENTED** [review]
:ship: 
/approve

**Review by MichaelClifford — APPROVED** [review]

**Pull request merged**